src/App.tsx

```diff
@@ -3,26 +3,57 @@ import ErrorPage from "./pages/ErrorPage/ErrorPage";
 import Home from "./pages/Home/Home";
 import "./styles/main.scss";
 import Calendar from "./pages/Calendar/Calendar";
-import { tribeUsers } from "./mockData/mockTribe";
 import ActiveTasks from "./pages/ActiveTasks/ActiveTasks";
 import Leaderboard from "./pages/Leaderboard/Leaderboard";
 import Profile from "./pages/Profile/Profile";
 import Login from "./pages/Login/Login";
 import Register from "./pages/Register/Register";
 import Account from "./pages/Account/Account";
 import { FirestoreProvider } from "./context/FirestoreProvider/FirestoreProvider";
-import { useState } from "react";
+import UpdateProfile from "./pages/UpdateProfile/UpdateProfile";
+import { useEffect, useState } from "react";
+import { UserProfile } from "./types/User";
+import { getDocumentFromFirestoreCollection } from "./utils/dbUtils";
+import { tribeUsers } from "./mockData/mockTribe";
 
 const App = () => {
   const [userUID, setUserUID] = useState<string>("1a38"); // Using Baheer as MOCK current user
   // NOTE: this console.log is used to workaround an eslint warning
   // It should be deleted once userUID is used
   console.log(userUID);
+  const [currentUser, setCurrentUser] = useState<UserProfile>({
+    id: "",
+    totalScore: 0,
+    name: "",
+    email: "",
+  });
+
+  const handleSetCurrentUser = (updatedUser: UserProfile) => {
+    setCurrentUser({ ...currentUser, ...updatedUser });
+  };
 
   const handleSetUserUID = (userUID: string | null) => {
     setUserUID(userUID || "");
   };
 
+  const fetchCurrentUser = async () => {
+    try {
+      const userRef = getDocumentFromFirestoreCollection(
```

Comment:
You can await the result and also use generics to pass the type to the function.

```ts
const userRef = await getDocumentFromFirestoreCollection<UserProfile>("test-tribe",
        "DrJZcEmb22Z5pG6fn2Fj2YYTHEy1"
      );
  
  if (userRef) {
    setCurrentUser(userData);
  }    
      
``

Reply:
done